dotCMS/src/main/java/com/dotcms/rest/ContentResource.java

```diff
@@ -1076,12 +1100,13 @@ protected void processMap(Contentlet contentlet, Map<String,Object> map) throws
 
 				// build a field map for easy lookup
 				Map<String,Field> fieldMap=new HashMap<String,Field>();
-				for(Field ff : FieldsCache.getFieldsByStructureInode(stInode))
+                List<Field> fields = new LegacyFieldTransformer(type.fields()).asOldFieldList();
+				for(Field ff : fields)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "fields". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "fields" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)